source/ui/dataman/AMScanView.cpp

```diff
@@ -575,13 +617,39 @@ MPlotGW * AMScanViewInternal::createDefaultPlot()
 
 	rv->plot()->setMarginRight(rv->plot()->marginLeft());
 
-	// DragZoomerTools need to be added first ("on the bottom") so they don't steal everyone else's mouse events
-	rv->plot()->addTool(new MPlotDragZoomerTool);
+	return rv;
+}
+
+void AMScanViewInternal::addToolToPlot(MPlot *plot, MPlotAbstractTool *tool)
+{
+	if (plot && tool) {
+		plot->addTool(tool);
 
-	// this tool adds mouse-wheel based zooming
-	rv->plot()->addTool(new MPlotWheelZoomerTool);
+		MPlotDataPositionTool *positionTool = qobject_cast<MPlotDataPositionTool*>(tool);
 
-	return rv;
+		if (positionTool) {
+			positionTool->setDataPositionIndicator(plot->axisScaleBottom(), plot->axisScaleLeft());
+		}
+	}
+}
+
+void AMScanViewInternal::removeToolFromPlot(MPlot *plot, MPlotAbstractTool *tool)
+{
+	if (plot && tool && plot->tools().contains(tool)) {
+		plot->removeTool(tool);
```

Comment:
I've modified the destructors for all AMScanViewInternal subclasses to call deleteLater() on all tools they've constructed.